converters/toZigbee.js

```diff
@@ -2295,25 +2333,45 @@ const converters = {
             if (typeof value === 'number') {
                 payload.brightness = value;
             }
-            if (meta.state.tuyaMode === 0 && payload.brightness) {
-                await entity.command('genLevelCtrl',
-                    'moveToLevel',
-                    {transtime: 0, level: payload.brightness},
-                    {disableResponse: true, disableDefaultResponse: true});
-                await entity.command('lightingColorCtrl', 'tuyaRgbMode', {enable: 0}, {}, {disableDefaultResponse: true});
-                return {state: {brightness: payload.brightness}};
+
+            if (meta.message.hasOwnProperty('color')) {
+                if (meta.message.color.h) {
+                    payload.hue = utils.mapNumberRange(meta.message.color.h, 0, 360, 0, 254);
+                }
+                if (meta.message.color.s) {
+                    payload.saturation = utils.mapNumberRange(meta.message.color.s, 0, 100, 0, 254);
+                }
+                if (meta.message.color.b) {
+                    payload.brightness = meta.message.color.b;
+                }
+                if (meta.message.color.brightness) {
+                    payload.brightness = meta.message.color.brightness;
+                }
             }
 
-            // if key is color -> make sure to switch to rgb mode
             await entity.command('lightingColorCtrl', 'tuyaRgbMode', {enable: 1}, {}, {disableDefaultResponse: true});
-            await entity.command('lightingColorCtrl', 'tuyaMoveToHueAndSaturationBrightness', payload, {disableDefaultResponse: true});
-            // transtime cannot be set on these devices. They seem to have a default one of about 1500ms!
-            return {state: {color_temp: value, brightness: payload.brightness}, readAfterWriteTime: payload.transtime * 100};
+            await entity.command('lightingColorCtrl', 'tuyaMoveToHueAndSaturationBrightness',
+                payload, utils.getOptions(meta.mapped, entity));
+
+            globalStore.putValue(entity, 'brightness', payload.brightness);
+
+            return {
```

Comment:
Can you also add the colorsync here? Example: https://github.com/Koenkk/zigbee-herdsman-converters/blob/932669961aa1f8c6d0735722178350eb884a9a3a/converters/fromZigbee.js#L564

Info about this: https://www.zigbee2mqtt.io/devices/81809_81813.html#notes